pkg/util/runtimeconfig/manager.go

```diff
@@ -73,19 +89,26 @@ func New(cfg Config, registerer prometheus.Registerer, logger log.Logger) (*Mana
 			Name: "runtime_config_hash",
 			Help: "Hash of the currently active runtime config file.",
 		}, []string{"sha256"}),
-		logger: logger,
+		logger:              logger,
+		bucketClientFactory: factory,
```

Comment:
Question, do we really need the `factory` variable? We can also just create `bucketClient` here directly?

Reply:
We can create the bucketClient here but I found that it's easier to mock the client in the unit tests if we create a factory. I saw that we do something similar for mocking the [bucket client in the compactor](https://github.com/cortexproject/cortex/blob/master/pkg/compactor/compactor.go#L340-L342).